homeassistant/components/zwave_js/discovery.py

```diff
@@ -1300,21 +1300,49 @@ def async_discover_single_value(
             continue
 
         # check device_class_generic
+        # If the value has an endpoint but it is missing on the node
+        # we can't match the endpoint device class to the schema device class.
+        # This could happen if the value is discovered after the node is ready.
```

Comment:
I added logic to handle this case since we had a test that discovered a value with an endpoint that wasn't in the node data.

We could remove this if that's not possible, but it seems it could happen according to Dominic.